openBCISample.js

```diff
@@ -78,20 +106,80 @@ module.exports = {
         }
     },
-    impedanceCalculation: function(sampleObject) {
+    /**
+     * @description Convert float number into three byte buffer. This is the opposite of .interpret24bitAsInt32()
+     * @param float - The number you want to convert
+     * @returns {Buffer} - 3-byte buffer containing the float
+     */
+    floatTo3ByteBuffer: function(float) {
+        var intBuf = new Buffer(3); // 3 bytes for 24 bits
+        intBuf.fill(0); // Fill the buffer with 0s
+
+        var temp = float / SCALE_FACTOR_CHANNEL; // Convert to counts
+
+        temp = Math.floor(temp); // Truncate counts number
+
+        // Move into buffer
+        intBuf[2] = temp & 255;
+        intBuf[1] = (temp & (255 << 8)) >> 8;
+        intBuf[0] = (temp & (255 << 16)) >> 16;
+
+        return intBuf;
+    },
+    /**
+     * @description Convert float number into three byte buffer. This is the opposite of .interpret24bitAsInt32()
+     * @param float - The number you want to convert
+     * @returns {Buffer} - 3-byte buffer containing the float
+     */
+    floatTo2ByteBuffer: function(float) {
+        var intBuf = new Buffer(2); // 2 bytes for 16 bits
+        intBuf.fill(0); // Fill the buffer with 0s
+
+        var temp = float / SCALE_FACTOR_ACCEL; // Convert to counts
+
+        temp = Math.floor(temp); // Truncate counts number
+
+        //console.log('Num: ' + temp);
+
+        // Move into buffer
+        intBuf[1] = temp & 255;
+        intBuf[0] = (temp & (255 << 8)) >> 8;
+
+        return intBuf;
+    },
+    /**
+     * Purpose: Calculate the impedance for one channel only.
+     * @param sampleObject - Standard OpenBCI sample object
+     * @param channelNumber - Number, the channel you want to calculate impedance for.
+     * @returns {Promise} - Fullfilled with impedance vaule for the specified channel.
+     * Author: AJ Keller
+     */
+    impedanceCalculationForChannel: function(sampleObject,channelNumber) {
         const sqrt2 = Math.sqrt(2);
         return new Promise((resolve,reject) => {
             if(sampleObject === undefined || sampleObject === null) reject('Sample Object cannot be null or undefined');
             if(sampleObject.channelData === undefined || sampleObject.channelData === null) reject('Channel cannot be null or undefined');
+            if(channelNumber < 1 || channelNumber > k.OBCINumberOfChannelsDefault) reject('Channel number invalid.');
+
+            var index = channelNumber - 1;
 
-            var impedanceArray = [];
-            for (var i = 1; i <= k.OBCINumberOfChannelsDefault; i++) {
-                impedanceArray[i] = (sqrt2 * sampleObject.channelData[i]) / k.OBCILeadOffDriveInAmps;
+            if (sampleObject.channelData[index] < 0) {
+                sampleObject.channelData[index] *= -1;
             }
-            sampleObject.impedanceArray = impedanceArray;
-            resolve(sampleObject);
+            var impedance = (sqrt2 * sampleObject.channelData[index]) / k.OBCILeadOffDriveInAmps;
```

Comment:
ok so you can sampleObject is sent into this method as input. This method is called in `._processBytes()` every time a 33-byte raw packets is converted to a sample object AND we are calculating impedance.